addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -346,6 +363,40 @@ def _get_l10n_in_edi_line_details(self, index, line, line_tax_details, sign):
         }
 
     def _l10n_in_edi_generate_invoice_json(self, invoice):
+        def put_discount_on(discount_line_vals, other_line_vals):
+            discount = max_discount = discount_line_vals['AssAmt'] * -1
+            if other_line_vals['AssAmt'] < discount:
```

Comment:
@jva-odoo I still think there are too many holes left in the way this is written. 

- Here, already there is no float_compare.  
- discount and max_discount are confusing.  Maybe from_discount and to_discount or something (just a wild idea)
- I would deduct the discount_line_vals first with the vals of other_line_vals, so you do that with the original amounts and only if there is still discount left (because you can introduce errors there as well)
- You actually do not need ratio_of_discount: either you discount the discount amount or yourself to 0

So, it is a matter of just comparing other_line_vals['AssAmt'] vs discount_line_vals['AssAmt'] and talking the smaller (abs) one.  (and either setting a line to 0 or subtracting the amount from the original lines)

I think it is simpler and will avoid lots of rounding errors.

Reply:
@jco-odoo

point 1:
okay I use float compare

point 2:
discount mins here that we need to put some amount as a discount but if the line value is less than the discount value then we need to calculate max_discount so I use this. if it's confusing then I change it no problem.

point 3: okay

point 4: okay got it.